python/GafferImageTest/CatalogueTest.py

```diff
@@ -64,6 +64,24 @@ def sendImage( image, catalogue, extraParameters = {}, waitForSave = True, close
 
 		return result
 
+	__catalogueIsRenderingMetadataKey = "gaffer:isRendering"
+
+	# Ignores Catalogue inserted data metadata when comparing images with ignoreMetadata = False.
+	def assertCatalogueImagesEqual( self, imageA, imageB, **kw ) :
+
+		ignoreMetadata = kw.get( "ignoreMetadata", False )
+		kw["ignoreMetadata"] = True
+
+		GafferImageTest.ImageTestCase.assertImagesEqual( self, imageA, imageB, **kw )
+
+		if not ignoreMetadata :
+			metaA = imageA[ "metadata" ].getValue()
+			metaB = imageB[ "metadata" ].getValue()
+			for m in ( metaA, metaB ) :
+				if self.__catalogueIsRenderingMetadataKey in m :
+						del m[ self.__catalogueIsRenderingMetadataKey ]
```

Comment:
I wonder if our future selves might thank us if we made this a feature of `assertImagesEqual()`. Perhaps we could just allow `ignoreMetadata = { "this", "that", "theOther" }` in addition to `ignoreMetadata = True`? That would also mean we'd implement it by testing a field at a time, which could give us much more helpful error messages.

Reply:
Haha, I was going to do that,  but then thought we'd get sidetracked on the interaction of the two option. Will do.

Reply:
We didn't need to change the comparison tests in the end.